specs/kvstore/WriterFencing.fizz

```diff
@@ -0,0 +1,145 @@
+# Model to validate that there can only be one writer and that old writers can be fenced.
+SUCCESS = "success"
+CONFLICT = "conflict"
+MAX_FLUSH_COUNT = 3
+NUM_WRITERS = 2
+
+role ObjectStore:
+  action Init:
+    self.objects = {}
+
+  atomic func ConditionalWrite(name, content):
+    if name in self.objects.keys():
+        return CONFLICT
+    self.objects[name] = content
+    return SUCCESS
+
+  atomic func Read(name):
+    return self.objects.get(name)
+
+  atomic func List():
+    return sorted(self.objects.keys())
+
+symmetric role SlateDBWriter:
+    action Init:
+        self.state = "waiting"
+        self.flush_count = 0
+        self.was_ever_open = 0
+    
+    atomic action FlushWal:
+        if self.state != "opened" or self.flush_count > MAX_FLUSH_COUNT:
```

Comment:
Also, if you could use the yaml frontmatter to configure an upper limit on the flushwal action, it would be cleaner as it flush_count is not part of the `specication` but used only to limit the model checker.

```
---
action_options:
    SlateDBWriter.FlushWal:
        max_actions: 3
---